app/src/main/java/com/infomaniak/mail/utils/Extensions.kt

```diff
@@ -275,6 +278,18 @@ fun Fragment.safeNavigateToNewMessageActivity(
 fun Fragment.safeNavigateToNewMessageActivity(args: Bundle? = null, currentClassName: String? = null) {
     if (canNavigate(currentClassName)) (requireActivity() as MainActivity).navigateToNewMessageActivity(args)
 }
+
+fun Fragment.navigateToDownloadProgressDialog(
+    attachment: Attachment,
+    attachmentIntentType: AttachmentIntentType,
+    currentClassName: String = AttachmentActionsBottomSheetDialog::class.java.name,
+) {
+    safeNavigate(
+        resId = R.id.downloadAttachmentProgressDialog,
+        args = attachment.createDownloadDialogNavArgs(attachmentIntentType),
+        currentClassName = currentClassName,
+    )
+}
```

Comment:
Do we still want to add more extensions to that file ? Maybe in a `NavigationExtensions` file ? We also need to move all `safeNavigate` extensions there.

Reply:
You're right, I added there because of habit and to put it with the other navigateToX() functions, I'll move them all in a new File